Rename DisplayList::DisplayListItem to DisplayList::Item and move it to a separate file

shallawa opened the pull request:
#### 46e9c071bff4d625b0d214b50e06c48c5ce542fa
<pre>
Rename DisplayList::DisplayListItem to DisplayList::Item and move it to a separate file
<a href="https://bugs.webkit.org/show_bug.cgi?id=260668">https://bugs.webkit.org/show_bug.cgi?id=260668</a>
rdar://114391530

Reviewed by Kimmo Kinnunen.

Remove the redundancy in the name and create dedicated source and header files
for the DisplayList::Item. Move its existing functions from DisplayListItems.cpp
and DisplayListReplayer.cpp to the new files.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/displaylists/DisplayList.cpp:
(WebCore::DisplayList::DisplayList::append):
(WebCore::DisplayList::DisplayList::asText const):
(WebCore::DisplayList::DisplayList::dump const):
* Source/WebCore/platform/graphics/displaylists/DisplayList.h:
(WebCore::DisplayList::DisplayList::items const):
(WebCore::DisplayList::DisplayList::items):
* Source/WebCore/platform/graphics/displaylists/DisplayListItem.cpp: Copied from Source/WebCore/platform/graphics/displaylists/DisplayListReplayer.cpp.
(WebCore::DisplayList::isValid):
(WebCore::DisplayList::applyImageBufferItem):
(WebCore::DisplayList::applyNativeImageItem):
(WebCore::DisplayList::applySourceImageItem):
(WebCore::DisplayList::applySetStateItem):
(WebCore::DisplayList::applyDrawGlyphs):
(WebCore::DisplayList::applyDrawDecomposedGlyphs):
(WebCore::DisplayList::applyItem):
(WebCore::DisplayList::shouldDumpItem):
(WebCore::DisplayList::dumpItem):
(WebCore::DisplayList::operator&lt;&lt;):
* Source/WebCore/platform/graphics/displaylists/DisplayListItem.h: Added.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::shouldDumpDisplayListItem): Deleted.
(WebCore::DisplayList::dumpDisplayListItem): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::isValid): Deleted.
(WebCore::DisplayList::operator&lt;&lt;): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.h:
(WebCore::DisplayList::RecorderImpl::append):
* Source/WebCore/platform/graphics/displaylists/DisplayListReplayer.cpp:
(WebCore::DisplayList::Replayer::Replayer):
(WebCore::DisplayList::Replayer::replay):
(WebCore::DisplayList::applyImageBufferItem): Deleted.
(WebCore::DisplayList::applyNativeImageItem): Deleted.
(WebCore::DisplayList::applySourceImageItem): Deleted.
(WebCore::DisplayList::applySetStateItem): Deleted.
(WebCore::DisplayList::applyDrawGlyphs): Deleted.
(WebCore::DisplayList::applyDrawDecomposedGlyphs): Deleted.
(WebCore::DisplayList::Replayer::applyItem): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListReplayer.h:
* Tools/TestWebKitAPI/Tests/WebCore/cg/DisplayListTestsCG.cpp:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/267431@main">https://commits.webkit.org/267431@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/18975f4b6ae63e1cfa07493132b1f7659c3f5f48

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16543 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16866 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17307 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18325 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15515 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16733 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20096 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17007 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17855 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16741 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17145 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14312 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19107 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14389 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14996 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/21786 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15377 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15162 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19472 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15756 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/13377 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14956 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3967 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19325 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15581 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple reviewed (approved):
I'd perhaps rather like to see removal of WebCore::DisplayList, as it's not necessarily a feature big enough to warrant a new namespace compared to all other features not having their own namespace. That being said, it is somewhat orthogonal to this patch.

shallawa committed:
Rename DisplayList::DisplayListItem to DisplayList::Item and move it to a separate file
https://bugs.webkit.org/show_bug.cgi?id=260668
rdar://114391530

Reviewed by Kimmo Kinnunen.

Remove the redundancy in the name and create dedicated source and header files
for the DisplayList::Item. Move its existing functions from DisplayListItems.cpp
and DisplayListReplayer.cpp to the new files.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/displaylists/DisplayList.cpp:
(WebCore::DisplayList::DisplayList::append):
(WebCore::DisplayList::DisplayList::asText const):
(WebCore::DisplayList::DisplayList::dump const):
* Source/WebCore/platform/graphics/displaylists/DisplayList.h:
(WebCore::DisplayList::DisplayList::items const):
(WebCore::DisplayList::DisplayList::items):
* Source/WebCore/platform/graphics/displaylists/DisplayListItem.cpp: Copied from Source/WebCore/platform/graphics/displaylists/DisplayListReplayer.cpp.
(WebCore::DisplayList::isValid):
(WebCore::DisplayList::applyImageBufferItem):
(WebCore::DisplayList::applyNativeImageItem):
(WebCore::DisplayList::applySourceImageItem):
(WebCore::DisplayList::applySetStateItem):
(WebCore::DisplayList::applyDrawGlyphs):
(WebCore::DisplayList::applyDrawDecomposedGlyphs):
(WebCore::DisplayList::applyItem):
(WebCore::DisplayList::shouldDumpItem):
(WebCore::DisplayList::dumpItem):
(WebCore::DisplayList::operator<<):
* Source/WebCore/platform/graphics/displaylists/DisplayListItem.h: Added.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::shouldDumpDisplayListItem): Deleted.
(WebCore::DisplayList::dumpDisplayListItem): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::isValid): Deleted.
(WebCore::DisplayList::operator<<): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.h:
(WebCore::DisplayList::RecorderImpl::append):
* Source/WebCore/platform/graphics/displaylists/DisplayListReplayer.cpp:
(WebCore::DisplayList::Replayer::Replayer):
(WebCore::DisplayList::Replayer::replay):
(WebCore::DisplayList::applyImageBufferItem): Deleted.
(WebCore::DisplayList::applyNativeImageItem): Deleted.
(WebCore::DisplayList::applySourceImageItem): Deleted.
(WebCore::DisplayList::applySetStateItem): Deleted.
(WebCore::DisplayList::applyDrawGlyphs): Deleted.
(WebCore::DisplayList::applyDrawDecomposedGlyphs): Deleted.
(WebCore::DisplayList::Replayer::applyItem): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListReplayer.h:
* Tools/TestWebKitAPI/Tests/WebCore/cg/DisplayListTestsCG.cpp:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/267431@main
